src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -2254,9 +2254,17 @@ GenerateGrantOnFunctionQueriesFromAclItem(Oid functionOid, AclItem *aclItem)
 		}
 		else
 		{
-			ereport(ERROR, (errmsg("unsupported prokind"),
-							errdetail("GRANT commands on procedures are propagated only "
-									  "for procedures and functions.")));
+			if (EnableWarningsPropagation)
+			{
+				ereport(ERROR, (errmsg("unsupported prokind"),
```

Comment:
I think we should NOT guard any ERRORs with this GUC.

Can we fix this error without the GUC?

Reply:
I agree, we should not guard the error to not change citus behaviour, so I removed that change from this PR. I will solve it on future sub PRs.